virtualizarr/writers/icechunk.py

```diff
@@ -23,7 +24,10 @@
 
 
 def dataset_to_icechunk(
-    ds: Dataset, store: "IcechunkStore", append_dim: Optional[str] = None
+    ds: Dataset,
+    store: "IcechunkStore",
+    append_dim: Optional[str] = None,
+    last_updated_at: Optional[datetime] = None,
```

Comment:
The type here is `datetime` or `None`, but the type upstream is more general. Should we also validate the input type here so that the user doesn't e.g. unwittingly pass a date as a string and be suprised when it doesn't behave the way they expected?